src/Provider/Photon/Photon.php

```diff
@@ -177,6 +177,22 @@ public function getName(): string
         return 'photon';
     }
 
+    private function buildLayerFilterQuery(mixed $layers): string
+    {
+        if (!is_iterable($layers)) {
+            $layers = [$layers];
+        }
+
+        if (!is_array($layers)) {
+            $layers = iterator_to_array($layers);
+        }
+
+        return implode('', array_map(
+            static fn ($layer) => sprintf('&layer=%s', $layer),
+            array_filter($layers, static fn ($layer) => is_scalar($layer)),
+        ));
+    }
```

Comment:
I find the way this function is written a bit confusing.
We indeed need to be able to process both `string` and `array` to avoid having a breaking change.

Couldn't we replicate what we did in `buildOsmTagFilterQuery()` function ?

```suggestion
    /**
     * @param string|string[] $layers
     */
    private function buildLayerFilterQuery(mixed $layers): string
    {
        if (is_string($layers)) {
            return '&layer='.urlencode($layers);
        }
        
        $query = '';
        foreach ($layersas $layer) {
            $query .= '&layer='.urlencode($layer);
        }
        return $query;
    }
```